README.md
zantasticox
============
for reasons of simplicity the deposit was transferred to bitbucket https://bitbucket.org/andykimpe/zantasticox/

Pour des raison de simplicité le depot à été tranférer sur bitbucket https://bitbucket.org/andykimpe/zantasticox/


changelog.txt
// Zantastico X Open Source Script Installer for ZPX
// Version          : 10.1.2.4 (12-23-2013)
// Updated By       : TGates for ZPanel X translation system andykimpe
// Additional Work  : Durandle
// Packages Updated : 03/08/2013 by jacobg830
// Contact Email    : tgates@zpanelcp.com
// Original Authors : Bobby Allen/Mudasir Mirza
 
09/10/2012 Zantastico X v1.0.0 released:
Initial release.

09/11/2012 v1.0.2 released:
Fixed wrong variable being used for installation link - Thanks to NightDragon
Auto-update feature removed. (More testing required)

09/13/2012 v10.0.5 released:
Fixed missing file error. - DaveTom13
Updated version number to coincide with ZPX
Added translation code
Worked on auto-updater
Edited layout

10/8/2012 v10.0.5.3 released:
Fixed layout issue in CentOS 6.3

10/19/2012 v10.0.5.4 released:
Added root install capability.
Better way of notifying users if a package needs a database.

Updated root install code
Added function to remove all files/folders in root folder
Added function to get server OS type and chmod/chown files/folders properly - still working on permissions
Added ImpressPages123 as a default package

Revision: 73
Author: jacobg830
Date: Wednesday, March 06, 2013 3:39:29 PM
Message:
Packages were updated to latest stable releases

Revision: 84
Author: tgates
Date: Tuesday, April 09, 2013 10:56:11 AM
Message:
added Dan's permissions code in code/functions.php;
Changed update URL back to zppy-repo.mach-hosting.com from modules.zpanelcp.com/repo/ (zpanel repo not fully working yet);

Revision: 91
Author: tgates
Date: Friday, August 02, 2013 1:43:55 PM
Message:
worked on 10.1.0 bootstrap compatibility (theming);

Revision: 94
Author: tgates
Date: Friday, August 02, 2013 7:36:33 PM
Message:
Added ability to use/add custom installers;
zanx-install.php - custom installer file;
if it exists, the code inside it will be run;

Revision: 96
Author: tgates
Date: Friday, August 02, 2013 10:03:25 PM
Message:
updated version number to 10.1.1.0;
added PNProv101;
added Moodle v251;
added XML tag <ostype> for Repo_Browser integration

Revision: 104
Author: jacobg830
Date: Saturday, August 03, 2013 2:11:26 PM
Message:
updated version number to 10.1.1.1;
Updated all packages

Revision: 107
Author: tgates
Date: Thursday, August 08, 2013 10:27:46 AM
Message:
updated version number to 10.1.1.2;
Fixed  minor issue with version/date time stamping;
Fixed header appearance (had wrong css variables);

Revision: 114
Author: tgates
Date: Thursday, September 05, 2013 5:38:15 PM
Message:
updated version number to 10.1.2.2;
Added requires database and user questions with links;
fixed joomla315.zip (files were missing);
minor fixes to installer code;

Revision: 116
Author: tgates
Date: Friday, September 13, 2013 2:19:03 PM
Message:
updated version number to 10.1.2.3;
updated Platinum Nuke Pro packages (pnp101.zip);
changed all repo references to ZPanel Module Repository;

Revision: 117
Author: andykimpe
Date: Monday, December 23, 2013 5:00:00 PM
Message:
updated version number to 10.1.2.4;
updated Joomla v 3.2 and all system language
updated Wordpress v 3.8 English and French Language
updated Drupal v 7.24 all system language
updated MyBB Forum v 1.6.11 all system language
updated phpBB v 3.0.12 all system language
updated Simple Machines Forum v 2.0.6 all system language
updated b2evolution v 5.0.6 all system language
updated Moodle v 2.6+ all system language
updated DokuWiki v 2013-12-08 "Binky" all system language
add zantasticotx full traslation system
add zantasticotx full french translation (updated on zpxfrtrad)


code/functions.php
<?php
// Zantastico X Open Source Script Installer for ZPX
// Version          : 10.1.2.4 (12-23-2013)
// Updated By       : TGates for ZPanel X translation system andykimpe
// Additional Work  : Durandle
// Packages Updated : 03/08/2013 by jacobg830
// Contact Email    : tgates@zpanelcp.com
// Original Authors : Bobby Allen/Mudasir Mirza

include('../../../cnf/db.php');
include('../../../inc/dbc.inc.php');

// Listing domains in DropDown Menu
function ListDomain($uid){
    global $zdbh;
    $sql="SELECT * FROM x_vhosts WHERE vh_acc_fk ='".$uid."' and vh_active_in='1' and vh_deleted_ts is NULL";
    $numrows = $zdbh->query($sql);
    if (@mysql_num_rows($numrows) == 0) {
        $sql = $zdbh->prepare($sql);
        $html="";
        $html .="<select name=site_domain>";
        $sql->execute();
        while ($rowsettings = $sql->fetch()) {
            $domain = $rowsettings['vh_name_vc'];
            $html .= "<option value=\"".$domain."\">".$domain."</option>";
        }
        $html .= "</select>";
        return $html;
    } else {
        echo "Unable to fetch domain list";
    }
}

function FetchDomainDir($uid,$domain){
    global $zdbh;
    $sql="SELECT * FROM x_vhosts WHERE vh_acc_fk='" . $uid . "' AND vh_name_vc='" . $domain . "'";
    $numrows = $zdbh->query($sql);
    if (@mysql_num_rows($numrows) == 0) {
        $sql = $zdbh->prepare($sql);
        $sql->execute();
        while ($rowsettings = $sql->fetch()) {
            $domaindir = $rowsettings['vh_directory_vc'];
        }
        return $domaindir;
    } else {
        echo "Unable to fetch domain dir";
        exit();
    }
}

// Function to create Directory
function CreateDir($dir){
    @$mkdir = mkdir($dir);
    if($mkdir){
        echo "<font color=\"green\">Folder created Successfully</font><br><br>";
    } else {
        echo "<font color=\"red\">Folder Already Exists.</font><br><br>";
    }
}

// Function to Unzip
function UnZip($zipfile,$dest_dir){
    $zip = new ZipArchive;
    $res = $zip->open($zipfile);
    if ($res === TRUE) {
	 $zip->extractTo($dest_dir);
	 $zip->close();
	 echo "<font color=\"green\">Unzip was successful</font><br><br>";
	 } else {
	 echo "<font color=\"red\">Unzip was not successful</font><br><br>";
    }
}

// Function to clean the User Input
function clean($var){
    $res=stripslashes(trim($var));
    return $res;
}

// Set file and folder permissions and ownership
function directoryToArray($directory, $recursive) {
    $array_items = array();
    if ($handle = opendir($directory)) {
        while (false !== ($file = readdir($handle))) {
            if ($file != "." && $file != "..") {
                if (is_dir($directory. "/" . $file)) {
                    if($recursive) {
                        $array_items = array_merge($array_items, directoryToArray($directory. "/" . $file, $recursive));
                    }
                    $file = $directory . "/" . $file;
                    $array_items[] = preg_replace("/\/\//si", "/", $file);
                } else {
                    $file = $directory . "/" . $file;
                    $array_items[] = preg_replace("/\/\//si", "/", $file);
                }
            }
        }
        closedir($handle);
    }
    return $array_items;
}
// Fix permissions of installed files since they will automatically be set to                       
// the apache user and group                                                                        
function fixPermissionsIssue($completedir){                                                         
    $sysOS = php_uname('s');                                                                        
    $zsudo = ctrl_options::GetOption('zsudo');                                                      
                                                                                                    
    switch($sysOS){                                                                                 
        case 'Linux':                                                                               
            exec("$zsudo chown -R ftpuser.ftpgroup " . $completedir);                               
            exec("$zsudo chmod -R 777 " . $completedir);                                            
        break;                                                                                      
        case 'Unix':                                                                                
            exec("$zsudo chown -R ftpuser:ftpgroup " . $completedir);                               
            exec("$zsudo chmod -R 777 " . $completedir);                                            
        break;                                                                                      
        default:                                                                                    
            //windows or uncompilable operating system !!Do Nothing!!                               
        break;                                                                                      
    }                                                                                               
}

// Clean out installation folder
function emptyDir($completedir) {
	$i = new DirectoryIterator($completedir);
		foreach($i as $f) {
			if($f->isFile()) {
				unlink($f->getRealPath());
			} else if(!$f->isDot() && $f->isDir()) {
				emptyDir($f->getRealPath());
		rmdir($f->getRealPath());
			}
		}
}

// Function for Database and Database user creation
function database($pkgInstall) {
	$dbcontent = "<div class=\"zgrid_wrapper\"><br /><br /><font color=\"red\"><strong>This package requires a database.</strong></font><br /><br />
        <strong>Have you created a new database for your installation? <font color=\"blue\">If not,</font> <a href=\"/?module=mysql_databases\" target=\"_blank\"><font color=\"blue\"><u>do so now!</u></strong></font></a><br /><br />
        <strong>Do not forget to also create a database user too! <a href=\"/?module=mysql_users\" target=\"_blank\"><font color=\"blue\"><u>Create a database user now!</u></strong></font></a><br />
 </div>";
	return $dbcontent;
}
?>


code/controller.ext.php
<?php
// Zantastico X Open Source Script Installer for ZPX
// Version          : 10.1.2.4 (12-23-2013)
// Updated By       : TGates for ZPanel X translation system andykimpe
// Additional Work  : Durandle
// Packages Updated : 03/08/2013 by jacobg830
// Contact Email    : tgates@zpanelcp.com
// Original Authors : Bobby Allen/Mudasir Mirza

// Function to retrieve remote XML for update check

function check_remote_xml($xmlurl,$destfile){
    $feed = simplexml_load_file($xmlurl);
    if ($feed)
    {
        // $feed is valid, save it
        $feed->asXML($destfile);
    } elseif (file_exists($destfile)) {
        // $feed is not valid, grab the last backup
        $feed = simplexml_load_file($destfile);
    } else {
        die('Unable to retrieve XML file');
    }
}

class module_controller {
    
    static $ok;

    static function getModuleVersion() {
        global $zdbh, $controller, $zlo;
        $module_path="./modules/" . $controller->GetControllerRequest('URL', 'module');
        
        // Get Update URL and Version From module.xml
        $mod_xml = "./modules/" . $controller->GetControllerRequest('URL', 'module') . "/module.xml";
        $mod_config = new xml_reader(fs_filehandler::ReadFileContents($mod_xml));
        $mod_config->Parse();
        $module_version = $mod_config->document->version[0]->tagData;
		echo " ".$module_version."";
    }
	
    static function getUpdateVersion() {
        global $zdbh, $controller, $zlo;
        $module_path="./modules/" . $controller->GetControllerRequest('URL', 'module');
        
        // Get Update URL and Version From module.xml
        $mod_xml = "./modules/" . $controller->GetControllerRequest('URL', 'module') . "/module.xml";
        $mod_config = new xml_reader(fs_filehandler::ReadFileContents($mod_xml));
        $mod_config->Parse();
        $module_updateurl = $mod_config->document->updateurl[0]->tagData;
        $module_version = $mod_config->document->version[0]->tagData;

        // Download XML in Update URL and get Download URL and Version
        $myfile = check_remote_xml($module_updateurl, $module_path."/zantasticox.xml");
        $update_config = new xml_reader(fs_filehandler::ReadFileContents($module_path."/zantasticox.xml"));
        $update_config->Parse();
        $update_url = $update_config->document->downloadurl[0]->tagData;
        $update_version = $update_config->document->latestversion[0]->tagData;
		echo " ".$update_version."";
    } 
	
    static function getCheckUpdate() {
        global $zdbh, $controller, $zlo;
        $module_path="./modules/" . $controller->GetControllerRequest('URL', 'module');
        
        // Get Update URL and Version From module.xml
        $mod_xml = "./modules/" . $controller->GetControllerRequest('URL', 'module') . "/module.xml";
        $mod_config = new xml_reader(fs_filehandler::ReadFileContents($mod_xml));
        $mod_config->Parse();
        $module_updateurl = $mod_config->document->updateurl[0]->tagData;
        $module_version = $mod_config->document->version[0]->tagData;

        // Download XML in Update URL and get Download URL and Version
        $myfile = check_remote_xml($module_updateurl, $module_path."/zantasticox.xml");
        $update_config = new xml_reader(fs_filehandler::ReadFileContents($module_path."/zantasticox.xml"));
        $update_config->Parse();
        $update_url = $update_config->document->downloadurl[0]->tagData;
        $update_version = $update_config->document->latestversion[0]->tagData;

        if($update_version > $module_version)
            return true;
        return false;
    }    
    
    static function doUpdateModule() {
        global $zdbh, $controller, $zlo;

        $myzsudo=ctrl_options::GetOption('zsudo');
        $zproot = ctrl_options::GetOption('zpanel_root');
        
        // Get Update URL and Version From module.xml
        $mod_xml = "./modules/" . $controller->GetControllerRequest('URL', 'module') . "/module.xml";
        $mod_config = new xml_reader(fs_filehandler::ReadFileContents($mod_xml));
        $mod_config->Parse();
        $module_updateurl = $mod_config->document->updateurl[0]->tagData;

        // Download XML in Update URL and get Download URL and Version
        exec("$myzsudo wget -O /tmp/".$controller->GetControllerRequest('URL', 'module').".xml $module_updateurl");
        sleep(2);
        exec("$myzsudo chmod 777 /tmp/".$controller->GetControllerRequest('URL', 'module').".xml");
        $update_xml = "/tmp/".$controller->GetControllerRequest('URL', 'module').".xml";
        $update_config = new xml_reader(fs_filehandler::ReadFileContents($update_xml));
        $update_config->Parse();
        $update_url = $update_config->document->downloadurl[0]->tagData;
        $file_name = basename($update_url);
        $folder_name= basename($update_url, ".zpp");
        
        if (fs_director::CheckFileExists("etc/zppy-cache/package-downloads/" . $file_name . "")) {
            exec("$myzsudo rm -f etc/zppy-cache/package-downloads/" . $file_name . "");
        }
        sleep(1);
        exec("$myzsudo wget -O " . $zproot . "etc/zppy-cache/package-downloads/" . $file_name . " ". $update_url ."");
        exec("$myzsudo chmod 777 " . $zproot . "etc/zppy-cache/package-downloads/" . $file_name . "");
       
        exec("$myzsudo mkdir /tmp/" . $folder_name . "");
        sleep(1);
        exec("$myzsudo unzip " . $zproot . "etc/zppy-cache/package-downloads/" . $folder_name . ".zpp  -d /tmp/" . $folder_name."");
        sleep(1);
        exec("$myzsudo \cp -fr /tmp/" . $folder_name. "/*  " . $zproot . "modules/" . $folder_name . "/");
        sleep(1);
        exec("$myzsudo chown -R apache.apache " . $zproot . "modules/" . $folder_name . "/");
        exec("$myzsudo chmod -R 777 " . $zproot . "modules/" . $folder_name . "/");
        sleep(1);
        exec("$myzsudo \rm -f /tmp/" . $folder_name . ".xml");
        exec("$myzsudo \rm -fr /tmp/" . $folder_name . "");
        }
		
    static function getCheckDBUpdates() {
        global $zdbh;
        include(ctrl_options::GetOption('zpanel_root') . '/cnf/db.php');

        // Updates
        $v_update_sql = $zdbh->prepare("UPDATE x_modules SET mo_version_in=10124 WHERE mo_name_vc='" . ui_module::GetModuleName() . "'");
        $v_update_sql->execute();
    }
   
    static function getModuleDesc() {
        $module_desc = ui_language::translate(ui_module::GetModuleDescription());
        return $module_desc;
    }

    static function getModuleName() {
        $module_name = ui_module::GetModuleName();
        return $module_name;
    }

    static function getModuleIcon() {
        global $controller;
        $module_icon = "./modules/" . $controller->GetControllerRequest('URL', 'module') . "/assets/icon.png";
        return $module_icon;
    }
	
	static function getPackageSelection() {
	$toReturn ="";
	
	//auto detect language user
	if (!isset($Langue)) {
    $Langue = explode(',',$_SERVER['HTTP_ACCEPT_LANGUAGE']);
    $Langue = strtolower(substr(chop($Langue[0]),0,2));
    }
	// intégrate with langue else use english
	if (file_exists('modules/zantasticox/packages/'.$Langue.'.packages.xml')) {
    $lang = $Langue;
    } else {
    $lang = en;
    } 
		$packages= new xml_reader(fs_filehandler::ReadFileContents('modules/zantasticox/packages/'.$lang.'.packages.xml'));
		$packages->Parse();

		$toReturn .="
		<table class=\"table table-striped\" border=\"0\" width=\"100%\">
		  <tr>
			<th>" . ui_language::translate( "Package" ) . "<br />
			" . ui_language::translate( "Name" ) . "</th>
			<th>" . ui_language::translate( "Version" ) . "<br />
			" . ui_language::translate( "Number" ) . "</th>
			<th>" . ui_language::translate( "Package" ) . "<br />
			" . ui_language::translate( "Type" ) . "</th>
			<th>" . ui_language::translate( "Package" ) . "<br />
			" . ui_language::translate( "Description" ) . "</th>
			<th>" . ui_language::translate( "Database" ) . "<br />
			" . ui_language::translate( "Required" ) . "?</th>
			<th>&nbsp;</th>
		  </tr>";
		foreach($packages->document->package as $package){
			// START - Info and DB tags by tgates
			if($package->db[0]->tagData=='yes') $package->pkgdb[0]->tagData="yes";
			else $package->pkgdb[0]->tagData="no";
			if($package->db[0]->tagData=='yes') $package->db[0]->tagData="<font color='green'><strong>" . ui_language::translate( "YES" ) . "</strong></font>";
			else $package->db[0]->tagData="<font color='red'><strong>" . ui_language::translate( "NO" ) . "</strong></font>";
			// END - Info and DB tags by tgates
		$toReturn .="<tr>
			<th>" .$package->name[0]->tagData. "</th>
			<th>" .$package->version[0]->tagData. "</th>
			<td>" .$package->type[0]->tagData. "</td>
			<td>" .$package->info[0]->tagData. "</td>
			<td><center>" .$package->db[0]->tagData. "</center></td>
			<td><a href=\"JavaScript:newPopup('modules/zantasticox/code/installer.php?pkgzip=" .$package->zipname[0]->tagData. "&pkg=" .$package->name[0]->tagData. "&pkgdb=" .$package->pkgdb[0]->tagData. "&startinstall=true');\"><button type=\"button\" class=\"btn btn-primary btn-small\">" . ui_language::translate( "Install" ) . "</button></a></td>
 			</tr>";
			}
		return $toReturn;
	}

	static function getCustomPackageSelection() {
		$toReturn ="";
		if (file_exists('modules/zantasticox/packages/custom_packages.xml')) {
		$packages= new xml_reader(fs_filehandler::ReadFileContents('modules/zantasticox/packages/custom_packages.xml'));
		$packages->Parse();

		foreach($packages->document->package as $package){
			// START - Info and DB tags by tgates
			if($package->db[0]->tagData=='yes') $package->pkgdb[0]->tagData="yes";
			else $package->pkgdb[0]->tagData="no";
			if($package->db[0]->tagData=='yes') $package->db[0]->tagData="<font color='green'><strong>" . ui_language::translate( "YES" ) . "</strong></font>";
			else $package->db[0]->tagData="<font color='red'><strong>" . ui_language::translate( "NO" ) . "</strong></font>";
			// END - Info and DB tags by tgates
		$toReturn .="<tr>
			<th>" .$package->name[0]->tagData. "</th>
			<th>" .$package->version[0]->tagData. "</th>
			<td>" .$package->type[0]->tagData. "</td>
			<td>" .$package->info[0]->tagData. "</td>
			<td><center>" .$package->db[0]->tagData. "</center></td>
			<td><a href=\"JavaScript:newPopup('modules/zantasticox/code/installer.php?pkgzip=" .$package->zipname[0]->tagData. "&pkg=" .$package->name[0]->tagData. "&pkgdb=" .$package->pkgdb[0]->tagData. "&startinstall=true');\"><button type=\"button\" class=\"btn btn-primary btn-small\">" . ui_language::translate( "Install" ) . "</button></a></td>
 			</tr>";
			}
			$toReturn .="</table><br />";
		} else { 
			$toReturn .="</table><br />";
		}
		return $toReturn;
	}
}
?>


code/installer.php
<?php
// Zantastico X Open Source Script Installer for ZPX
// Version          : 10.1.2.3 (09-13-2013)
// Updated By       : TGates for ZPanel X
// Additional Work  : Durandle
// Packages Updated : 03/08/2013 by jacobg830
// Contact Email    : tgates@zpanelcp.com
// Original Authors : Bobby Allen/Mudasir Mirza

$zipfile = $_GET["pkgzip"];
$pkgInstall = $_GET['pkg'];
$pkgdb = $_GET['pkgdb'];
$start = $_GET['startinstall'];

if($start){
include('../../../cnf/db.php');
include('../../../dryden/db/driver.class.php');
include('../../../dryden/debug/logger.class.php');
include('../../../dryden/runtime/dataobject.class.php');
include('../../../dryden/sys/versions.class.php');
include('../../../dryden/ctrl/options.class.php');
include('../../../dryden/ctrl/auth.class.php');
include('../../../dryden/ctrl/users.class.php');
include('../../../dryden/fs/director.class.php');
include('../../../inc/dbc.inc.php');
include('functions.php');

if(!isset($_POST['submit'])) {
  
session_start();
if (isset($_SESSION['zpuid'])){
$userid = $_SESSION['zpuid'];
$currentuser = ctrl_users::GetUserDetail($userid);
$hostdatadir = ctrl_options::GetOption('hosted_dir')."".$currentuser['username'];
$userName = $currentuser['username'];
$random=rand();
$sysOS=php_uname('s');
?>
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
<title>ZPanel &gt; Zantastico X Installer - <?php echo $pkgInstall ?></title>
<link href="../../../etc/styles/<?php echo $currentuser['usertheme']; ?>/css/<?php echo $currentuser['usercss']; ?>.css" rel="stylesheet" type="text/css">
<link href="../assets/install-form.css?<?php echo $random; ?>" rel="stylesheet" type="text/css">
<link href="../assets/tooltip.css?<?php echo $random; ?>" rel="stylesheet" type="text/css">
<script src="../assets/ajaxsbmt.js?<?php echo $random; ?>" type="text/javascript"></script>
<script src="http://code.jquery.com/jquery-latest.js"></script>
 
</head>
<body style="background: #F3F3F3; font-size:12px">

<div style="margin-left:20px;margin-right:20px;">
<h2>ZPanel Zantastico X - <?php echo $pkgInstall ?></h2>
<div id="RunSubmit" style="height:100%;margin:auto;">
<p>Please provide the domain and folder name to start the installation of <?php echo $pkgInstall ?>.</p>

<?php
if ($pkgdb == 'yes') {
database($pkgInstall);
} else {
}
?>

<div class="zgrid_wrapper"><br />
<?php if ($pkgdb == "yes") { ?>
	<table border="0" cellspacing="1" cellpadding="1" align="center" width="600px">
        <tr>
			<td>
                <font color="red"><strong>This package requires a database<br />and database user.</strong></font><br />
            </td>
            <td>
				Create a database now? <a target="_blank" href="../../../?module=mysql_databases">YES!</a><br />
            	Create a database user now? <a target="_blank" href="../../../?module=mysql_users">YES!</a><br />
            </td>
        </tr>
	</table>
    <br />
<?php } ?> 
<strong>Enter package install folder information:</strong><br /><br />
<form name="doInstall" action="installer.php?startinstall=true&u=<?php echo $currentuser['userid']; ?>&pkgzip=<?php echo $zipfile ?>&pkg=<?php echo $pkgInstall ?>" method="post" onsubmit="xmlhttpPost('installer.php?startinstall=true&u=<?php echo $currentuser['userid']; ?>&pkgzip=<?php echo $zipfile ?>&pkg=<?php echo $pkgInstall ?>&pkgdb=<?php echo $pkgdb ?>', 'doInstall', 'RunResult', 'Unpacking the files, please wait...<br /><img src=\'../assets/bar.gif\'>'); return false;">
<table border="0" cellspacing="1" cellpadding="1" align="center" width="600px">
        <tr>
            <td>
                <label for="site_domain">Domain to install to: </label>
            </td>
            <td align="left">
                <?php
                $list = ListDomain($currentuser['userid']);
                echo $list;
                ?>
            </td>
            <td>Server OS: <?php echo $sysOS; ?></td>
        </tr>
        <tr>
            <td>
                <label for="install_to_base_dir">Install into the selected<br />domain's root folder?</label>
            </td>
            <td>
                <input type="checkbox" name="install_to_base_dir" value="1"/>
            </td>
            <td>
            	<font color="red"><strong>ALL FILES AND FOLDERS<br />WILL BE DELETED!</strong></font>
            </td>
        </tr>
        <tr>
            <td>
                <label for="dir_to_install">Install Folder: </label>
            </td>
            <td align="left">
                <input type="text" length="50" maxsize="100" name="dir_to_install" /><br />
                (Leave blank if installing<br />to the root folder)
            </td>
            <td>
            	<font color="red"><strong>NOTE:</strong></font> No "/".
            </td>
        </tr>
        <tr>
        	<td></td>
            <td><br />
             <center><button class="fg-button ui-state-default ui-corner-all" id="SubmitRun" name="submit" align="center" type="submit" value="">Start Install</button></center><br />
             <center><input  class="fg-button ui-state-default ui-corner-all" type="button" align="center" value="Cancel" onclick="self.close()"></center>
          </td>
          <td></td>
       </tr>
    </table>
             
</form>
</div>
</div>
<div id="RunResult" style="display:block;height:100%;margin:auto;">
<br /><br />
Unpacking the <?php echo $pkgInstall ?> files, please wait...<br /><br /><img src='../assets/bar.gif'>
</div>
</div>
</body>
</html>
<?php
} else { ?>
<body style="background: #F3F3F3;">
<h2>Unauthorized Access!</h2>
You have no permission to view this module.
</body>
<?php 

}
}else {
    
    $userid=$_GET['u'];
    $currentuser=ctrl_users::GetUserDetail($userid);
    $hostdatadir = ctrl_options::GetOption('hosted_dir')."".$currentuser['username'];
           
    $site_domain=clean($_POST['site_domain']);
    $dir_to_install=clean($_POST['dir_to_install']);
    $install_to_base_dir=clean($_POST['install_to_base_dir']);
    
    // Retrieve the directory for the Domain selected
    $domaindir=FetchDomainDir($userid, $site_domain);
    
    $completedir=$hostdatadir . "/public_html" . $domaindir . "/" . $dir_to_install . "" ;
    
    echo "<br /><b>Automated " .$pkgInstall. " Installation Status:</b><br /><br />";
        if(file_exists($completedir) and $install_to_base_dir != "1") {
        echo "<div><font color=\"red\"><strong>Destination folder already exists!</strong></font><br /><br />Sorry, the install folder (<strong>/public_html" . $domaindir . "/" . $dir_to_install . "</strong>) already exists, please go back and create a new folder!<br />";
        echo "<p><center><form><input class=\"fg-button ui-state-default ui-corner-all\" type=\"button\" type=\"button\" onClick=\"history.go(0)\" value=\"Go Back\"></form></center></p></div>";
            } else {

    // Creating Directory
	CreateDir($completedir);
    sleep(1);
	// Remove all Files in the install Folder
	emptyDir($completedir);
	sleep(3);
    // Un-Compressing The ZIP Archive
	set_time_limit(300);
	UnZip("../packages/" .$zipfile. ".zip", $completedir);
    sleep(5); 
    // Set file/folder permissions if on *nix
	fixPermissionsIssue($completedir);
	sleep(5);
      
$domaindir = preg_replace('/_/', '.', $site_domain);
echo "Package unzipped to: http://" . $site_domain . "/" . $dir_to_install . "<br><br>";
echo "If no errors, click to continue: ";
		if(file_exists($completedir . "/zanx-install.php")) {
			    echo "<a target=\"_blank\" href='http://" . $site_domain . "/" . $dir_to_install . "/zanx-install.php'> <button class=\"fg-button ui-state-default ui-corner-all\" type=\"button\">Install</button> </a>"; '';
				echo "then <input  class=\"fg-button ui-state-default ui-corner-all\" type=\"button\" type=\"button\" value=\"Close this window\" onclick=\"self.close()\">";
			} else {
				echo "<a target=\"_blank\" href='http://" . $site_domain . "/" . $dir_to_install . "'><button class=\"fg-button ui-state-default ui-corner-all\" type=\"button\">Install</button> </a>";
				echo "then <input  class=\"fg-button ui-state-default ui-corner-all\" type=\"button\" type=\"button\" value=\"Close this window\" onclick=\"self.close()\">";
            }
			}
}

}else{
    echo "<font color=\"red\">Unable to start Install Process</font>";
    exit();
}

?>

<script type="text/javascript">
    $(document).ready(function() { 
        $("#RunResult").hide();
            $("#SubmitRun").click(function(){
            $("#RunSubmit").hide();
            $("#RunResult").show();
        }); 
    })
</script>
